### PR TITLE
HedgilV2 new tests

### PR DESCRIPTION
**Pull request by 16slim**
Added:
- Changed SpiritJoint to v1 wait for its own tests as it does not compile
- Bug from migrateProvider
- Included test to manually operate the joint and providers
- Included test to use the automatic epoch roll
- Minor bug fixes in other tests due to whales and balances changing

### CODE REVIEW

File: scripts/deploy_providers_and_joint.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    vaultB = Contract("0x0DEC85e74A92c52b7F708c4B10207D9560CEFaf0")
+
+    # Provider strats
+    if network.show_active() == "ftm-main":
+        provider_strat = acct.deploy(ProviderStrategy, vaultA, publish_source=True)
+    else:
+        provider_strat = acct.deploy(ProviderStrategy, vaultA)
+
+    print(f"Original provider strat deployed to {provider_strat.address}")
+    providerA = ProviderStrategy.at(provider_strat.clone(vaultA).return_value)
```
Comment (jmonteer) [inline scripts/deploy_providers_and_joint.py:21]: are you cloning the right strategy?

Comment (jmonteer) [reply scripts/deploy_providers_and_joint.py:21]: or you would need to deploy the first one

File: scripts/deploy_providers_and_joint.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    
+    # USDC
+    tokenA = Contract("0x04068DA6C83AFCFA0e13ba15A6696662335D5B75")
+    vaultA = Contract("0xEF0210eB96c7EB36AF8ed1c20306462764935607")
+    # WFTM
+    tokenB = Contract("0x21be370D5312f44cB42ce377BC9b8a0cEF1A4C83")
+    vaultB = Contract("0x0DEC85e74A92c52b7F708c4B10207D9560CEFaf0")
+
+    # Provider strats
+    if network.show_active() == "ftm-main":
```
Comment (jmonteer) [inline scripts/deploy_providers_and_joint.py:15]: beautiful

File: tests/utils/checks.py [hunk 3]
```diff
@@ -10,7 +10,8 @@ def check_vault_empty(vault):
 
 def epoch_started(providerA, providerB, joint, amountA, amountB):
     assert pytest.approx(providerA.estimatedTotalAssets(), rel=2e-3) == amountA
-    assert pytest.approx(providerB.estimatedTotalAssets(), rel=2e-3) == amountB
+    # Less precision toa ccount for hedgil cost!
+    assert pytest.approx(providerB.estimatedTotalAssets(), rel=5e-2) == amountB
```
Comment (jmonteer) [inline tests/utils/checks.py:14]: why different to providerA?

**Review by jmonteer — COMMENTED** [review]
awesome job

File: scripts/deploy_providers_and_joint.py (hunk 1 above)
Comment (16slim) [reply scripts/deploy_providers_and_joint.py:21]: I am deploying an original Provider -> provider_strat and then clone that one twice. I could use the original as the providerA or providerB but I like the original to remain unused

File: tests/conftest.py [hunk 4]
```diff
@@ -517,6 +517,9 @@ def joint(
         )
         joint.setMaxPercentageLoss(500, {"from": gov})
```
[the remark this answers is not in the document]
Comment (16slim) [reply tests/conftest.py:518]: No need, changing it

File: tests/utils/checks.py (hunk 3 above)
Comment (16slim) [reply tests/utils/checks.py:14]: Because providerB has to pay for hedgilV2 cost

File: scripts/deploy_providers_and_joint.py (hunk 2 above)
Comment (16slim) [reply scripts/deploy_providers_and_joint.py:15]: eso tú

File: scripts/deploy_providers_and_joint.py (hunk 1 above)
Comment (jmonteer) [reply scripts/deploy_providers_and_joint.py:21]: coolio

**Comment by jmonteer** [comment]
let's get this money hehe

**Pull request merged**